### PR TITLE
[MBL-17455][Student] Compose RCE

### PR DESCRIPTION
**Pull request by hermannakos**
Test plan: Test the RCE in the create ToDo screen. Smoke test other RCEs, the image insertion has changed a bit.

refs: MBL-17455
affects: Student
release note: none

## Checklist

- [x] Tested in dark mode
- [x] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/RCEControls.kt [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+            RCEColorControls(onColorClick)
+        }
+        Divider()
+    }
+}
+
+@Composable
+fun RCETextControls(rceState: RCEState, onActionClick: (RCEAction) -> Unit) {
+    LazyRow {
+        item {
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/RCEControls.kt:71]: Shouldn't we use `items` when adding multiple items to a row?

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/RCEControls.kt [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+@Composable
+fun RCEColorControls(onColorClick: (Int) -> Unit) {
+    val context = LocalContext.current
+    LazyRow(
+        modifier = Modifier
+            .height(48.dp)
+            .padding(vertical = 12.dp)
+    ) {
+        item {
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/RCEControls.kt:161]: Shouldn't we use `items` when adding multiple items to a row?

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/RCEControls.kt [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.instructure.pandautils.R
+
+@Composable
+fun RCEControls(
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/RCEControls.kt:52]: By our conventions we should add a Modifier as the first optional parameter to all of the Composables in this file.

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+import instructure.rceditor.RCETextEditor
+import jp.wasabeef.richeditor.RichEditor
+
+@Composable
+fun ComposeRCE(
+    hint: String,
+    html: String,
+    modifier: Modifier = Modifier,
+    canvasContext: CanvasContext = CanvasContext.defaultCanvasContext(),
+    onTextChangeListener: (String) -> Unit
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt:53]: I am not sure about this, but shouldn't mandatory parameters be before the optional ones?

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+    val imagePickerLauncher =
+        rememberLauncherForActivityResult(contract = ActivityResultContracts.GetContent()) {
+            it?.let { imageUri ->
+                MediaUploadUtils.uploadRceImageJob(
+                    imageUri,
+                    canvasContext,
+                    context.getFragmentActivity()
+                ) { imageUrl ->
+                    rceTextEditor.insertImage(imageUrl, "")
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt:86]: Why don't we add alt text for the image picker images? We do this in other places where we use RCE.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt (hunk 4 above)
Comment (hermannakos) [reply libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt:53]: it's so we could use lambdas

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt [hunk 6]
```diff
@@ -70,14 +59,21 @@ fun ComposeRCE(
     var rceState by remember { mutableStateOf(RCEState()) }
     var showControls by remember { mutableStateOf(false) }
 
+    var focused by remember { mutableStateOf(false) }
+
     val context = LocalContext.current
     var rceTextEditor = RCETextEditor(context).apply {
+        setEditorHeight(280.dp.value.toInt().toPx)
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/rce/ComposeRCE.kt:66]: You can simplify this to `280.toPx`, we have an extension like this for Int.

**Review by tamaskozmer — APPROVED** [review]

**Pull request merged**